ci/src/lib.rs

```diff
@@ -0,0 +1,345 @@
+            "Couldn't retrieve ssh-agent key #{}",
+            options.ssh_index
+        ))?;
+        agent.userauth(&options.ssh_user, identity)?;
+    }
+    if sess.authenticated() {
+        spinner.finish();
+    }
+
+    let verbosity = options.verbose;
+    let execute_fn = |cmd, spinner| execute_cmd_with_spinner(&sess, cmd, spinner, verbosity);
+
+    // Check requirements
+    term::blank();
+    term::info!("Checking requirements:");
+    term::blank();
+
+    let mut requirements = vec![
+        "wget",
+        "bash",
+        "cat",
+        "grep",
+        "cut",
+        "mkdir",
+        "docker",
+        "docker-compose",
+    ];
+    if options.config.is_some() {
+        requirements.remove(0);
+    }
+
+    if requirements
+        .iter()
+        .map(|req| {
+            let spinner = term::spinner(&format!(
+                "Checking if {} exists",
+                term::format::highlight(req)
+            ));
+            let (_, status) =
+                execute_cmd_with_spinner(&sess, &format!("which {}", req), spinner, verbosity);
+            status
+        })
+        .sum::<i32>()
+        > 0
+    {
+        term::info!("Requirements are not installed.");
+        std::process::exit(1);
+    }
+    term::blank();
+
+    // Apply changes
+    if options.install {
+        // Make sure directories exist
+        let spinner = term::spinner("Making sure directory /app/radicle/etc exists");
+        let (_, status) = execute_fn("mkdir -p /app/radicle/etc", spinner);
+        if status != 0 {
+            std::process::exit(1);
+        }
+
+        let spinner = term::spinner("Making sure directory /app/radicle/hooks exists");
+        let (_, status) = execute_fn("mkdir -p /app/radicle/hooks", spinner);
+        if status != 0 {
+            std::process::exit(1);
+        }
+        term::blank();
+
+        // Upload custom config file instead when supplied
+        if options.config.is_some() {
+            let spinner = term::spinner("Copying config yaml to server");
+            let path = options.config.unwrap();
+            let mut file = File::open(path).context("Couldn't open config file")?;
+            let mut content = String::new();
+            file.read_to_string(&mut content)
+                .context("Couldn't read the contents of config file")?;
+
+            upload_content_to_file(
+                &sess,
+                content.as_bytes(),
+                "/app/radicle/etc/concourse-docker-compose.yml",
+                0o644,
+            );
+            spinner.finish();
+        }
+
+        // Copy receive-hook file over
+        let spinner = term::spinner("Copying receive-hook file to server");
+        let receive_hook_content = include_str!("../receive-hook");
+        upload_content_to_file(
+            &sess,
+            receive_hook_content.as_bytes(),
+            "/app/radicle/hooks/receieve-hook",
+            0o755,
+        );
```

Comment:
This is not going to work, as you don't know how `--root` was configured on the server. I think it's best to either have a `--remote-root` flag to specify it, or to let users install this file manually.